src/docfx/lib/git/GitCommitProvider.cs

```diff
@@ -11,70 +11,66 @@ namespace Microsoft.Docs.Build
     internal sealed class GitCommitProvider : IDisposable
     {
         private readonly ConcurrentDictionary<string, Repository> _repositoryByFolder = new ConcurrentDictionary<string, Repository>();
-        private readonly ConcurrentDictionary<string, Lazy<Task<(FileCommitProvider provider, string repoRemote)>>> _fileCommitProvidersByRepoPath = new ConcurrentDictionary<string, Lazy<Task<(FileCommitProvider provider, string repoRemote)>>>();
+        private readonly ConcurrentDictionary<string, (FileCommitProvider provider, string repoRemote)> _fileCommitProvidersByRepoPath = new ConcurrentDictionary<string, (FileCommitProvider provider, string repoRemote)>();
```

Comment:
Shall we keep `lazy` here?
Let's say, `GetOrAdd` get invoked concurrently, only 1 `FileCommitProvider` been added to the dictionary, and the ones not been added have been instanced. How could we dispose them?

Reply:
good point here, let me change the code